gcloud/logging/config_service_v2_api.py

```diff
@@ -14,64 +14,123 @@
+from google.gax import config
+from google.gax.path_template import PathTemplate
+import google.gax
+import yaml
+
 from google.logging.v2 import logging_config_pb2
 
 
 class ConfigServiceV2Api(object):
+    """See src/api/google/logging/v2/logging.proto for documentation"""
 
-    # The default address of the logging service.
-    _SERVICE_ADDRESS = 'logging.googleapis.com'
+    _CODE_GEN_NAME_VERSION = 'gapic-0.1.0'
 
-    # The default port of the logging service.
-    _DEFAULT_SERVICE_PORT = 443
+    SERVICE_ADDRESS = 'logging.googleapis.com'
+    """The default address of the service."""
+
+    DEFAULT_SERVICE_PORT = 443
+    """The default port of the service."""
 
     # The scopes needed to make gRPC calls to all of the methods defined in
     # this service
-    _ALL_SCOPES = (
-        'https://www.googleapis.com/auth/logging.write',
-        'https://www.googleapis.com/auth/logging.admin',
-        'https://www.googleapis.com/auth/logging.read',
-        'https://www.googleapis.com/auth/cloud-platform.read-only',
-        'https://www.googleapis.com/auth/cloud-platform',
-    )
-
-    _LIST_SINKS_DESCRIPTOR = page_descriptor.PageDescriptor(
-        'page_token',
-        'next_page_token',
-        'sinks',
-    )
-
-    def __init__(
-            self,
-            service_path=_SERVICE_ADDRESS,
-            port=_DEFAULT_SERVICE_PORT,
-            channel=None,
-            ssl_creds=None,
-            scopes=_ALL_SCOPES,
-            is_idempotent_retrying=True,
-            max_attempts=3,
-            timeout=30):
-        self.defaults = api_callable.ApiCallableDefaults(
-            timeout=timeout,
-            max_attempts=max_attempts,
-            is_idempotent_retrying=is_idempotent_retrying)
-        self.stub = api_utils.create_stub(
+    _ALL_SCOPES = ('https://www.googleapis.com/auth/logging.write',
+                   'https://www.googleapis.com/auth/logging.admin',
+                   'https://www.googleapis.com/auth/logging.read',
+                   'https://www.googleapis.com/auth/cloud-platform.read-only',
+                   'https://www.googleapis.com/auth/cloud-platform', )
+
+    _PROJECT_PATH_TEMPLATE = PathTemplate('projects/{project}')
+    _SINK_PATH_TEMPLATE = PathTemplate('projects/{project}/sinks/{sink}')
+
+    @classmethod
+    def project_path(cls, project):
+        """Returns a fully-qualified project resource name string."""
+        return cls._PROJECT_PATH_TEMPLATE.instantiate({'project': project, })
+
+    @classmethod
+    def sink_path(cls, project, sink):
+        """Returns a fully-qualified sink resource name string."""
+        return cls._SINK_PATH_TEMPLATE.instantiate({
+            'project': project,
+            'sink': sink,
+        })
+
+    def __init__(self,
+                 service_path=SERVICE_ADDRESS,
+                 port=DEFAULT_SERVICE_PORT,
+                 channel=None,
+                 ssl_creds=None,
+                 scopes=_ALL_SCOPES,
```

Comment:
We should probably be careful regarding mutability here, right? If _ALL_SCOPES happened to be changed to a list at some point, this would be a potential issue.